### ISSUE TITLE
Base tools are not accessible to cornerstoneTools consumers

### ISSUE THREAD

**Issue by mikehazell** [cornerstonejs/cornerstoneTools#878]
### Prerequisites

* [x] Are you running the latest version?
* [x] Are you reporting to the correct repository?
* [x] Did you perform a cursory search?

### Description

The base tool classes (BaseAnnotationTool, BaseBrushTool, BaseTool) are not exported in `index.js`. This makes it really awkward to create custom tools as a consumer of cornerstoneTools.

### Steps to Reproduce

n/a

**Expected behavior:** [What you expected to happen]

Should be able to import base tools as a 3rd partly or consumer of the library.

```js
import { BaseAnnotationTool } from 'cornerstoneTools';
class MyCustomTool extends BaseAnnotationTool {
   // ...
}
```

**Actual behavior:** [What actually happened]

It's nasty, you need to dig though the prototype chain to find a BaseTool class. I'm sure this is just an oversight.

**Comment by dannyrb** [comment]
Hmm.. So we did this weird thing where "less public" bits aren't exposed as top level API members. To access them, you do something like:

```js
import { import as csTools } from 'cornerstone-tools';

const BaseTool = csTools('base/BaseTool');
```

You can see a list of those in the `lib.js` file: https://github.com/cornerstonejs/cornerstoneTools/blob/master/src/lib.js#L78-L80

Our goal was, and we may have missed the mark, to emulate what we saw in the [Quill.js Extensions](https://quilljs.com/docs/api/#extension). This would allow plugin authors to register their own code, extend cornerstone-tools, and leverage code registered by other plugin authors as well.

Top level API members feel like good "out of the box" features, where anyone doing a simple viewer may be satisfied with everything available there. Anything in `lib.js` is more for plugin authors, advanced features, and extending cornerstone-tools.

Saying all of that, I'm still not sure how I feel about the decision, and I'm open to suggestions. The, as yet, sparse documentation does not help matters. Regardless, the BaseTool's should probably be a bit more accessible and exposed; I'm still unsure about features like: `getRGBPixels`

**tl;dr:** I do think the BaseTool's can safely be added to top level exports, and that they should be.

**Comment by dannyrb** [comment]
@JamesAPetts, @mikehazell brings up a good point. This feels like something you'd want to have awareness of or discuss.

**Comment by mikehazell** [comment]
Interesting. That's not a pattern I've seen before. Certainly not something I would have guessed as a new user. That may be because my world is mostly React and this is a different community with different opinions and experience.

**Comment by dannyrb** [comment]
@mikehazell, we're very open to alternatives and other opinions. In particular, if you're aware of OSS projects that have plugins, extensions, modules, etc. I would enjoy taking a peak at their solutions so we can learn from them.

**Comment by mikehazell** [comment]
I see where I went wrong reading the docs. I was looking at "Custom Tools" and missed the line about the "third party tool guide".

It might be nice to flip the target audience of those docs and assume the reader is most likely a cornerstoneTools user rather than a contributor. I might have a look at that.

**Comment by dannyrb** [comment]
@mikehazell, I'm going to close this for now based on the discussion we've had. Please feel free to create a separate issue for updating the documents as an alternative solution.
